packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -53,3 +53,22 @@ export async function TraverseMainTspFileInWorkspace() {
         .map((uri) => normalizeSlashes(uri.fsPath)),
     );
 }
+
+type TspCliType = "tsp-npm-global" | "tsp-standalone" | "not-available";
+export async function checkTspCliType(): Promise<TspCliType> {
+  try {
+    // use "where" or "which" command instead of "tsp --version" which is much slower
+    const command = process.platform === "win32" ? "where" : "which";
+    const r = await spawnExecution(command, ["tsp"], process.cwd());
```

Comment:
could maybe use https://www.npmjs.com/package/which instead, its pretty small dependency and that would probably be faster/more reliable